File: core/classifications.py
def classify_who_standard(bmi: float) -> tuple[str, str]:
    if bmi < 18.5:
        return ("Underweight", "Low (but other risks)")
    elif bmi < 25.0:
        return ("Normal weight", "Average")
    elif bmi < 30.0:
        return ("Overweight", "Increased")
    elif bmi < 35.0:
        return ("Obese Class I", "High")
    elif bmi < 40.0:
        return ("Obese Class II", "Very High")
    else:
        return ("Obese Class III", "Extremely High")


def classify_who_asian(bmi: float) -> tuple[str, str]:
    if bmi < 18.5:
        return ("Underweight", "Low (but other risks)")
    elif bmi < 23.0:
        return ("Normal weight", "Average")
    elif bmi < 27.5:
        return ("Overweight", "Increased")
    else:
        return ("Obese", "High")


def classify_ponderal(pi: float) -> tuple[str, str]:
    if pi < 11.0:
        return ("Underweight", "Low")
    elif pi <= 14.0:
        return ("Normal", "Average")
    else:
        return ("Overweight", "Elevated")


def classify_new_bmi(new_bmi: float) -> tuple[str, str]:
    return classify_who_standard(new_bmi)


def classify_whtr(whtr: float) -> tuple[str, str]:
    if whtr < 0.4:
        return ("Extremely Slim", "Low (but other risks)")
    elif whtr < 0.5:
        return ("Healthy", "Average")
    elif whtr < 0.6:
        return ("Increased Risk", "Increased")
    else:
        return ("High Risk", "High")


def get_risk_color(risk: str) -> str:
    mapping = {
        "Low (but other risks)": "blue",
        "Average": "green",
        "Increased": "orange",
        "High": "red",
        "Very High": "red",
        "Extremely High": "red",
        "Low": "blue",
        "Elevated": "orange",
    }
    return mapping.get(risk, "gray")

File: mobile/lib/core/classifications.dart
typedef Classification = (String category, String risk);

Classification classifyWhoStandard(double bmi) {
  if (bmi < 18.5) return ('Underweight', 'Low (but other risks)');
  if (bmi < 25.0) return ('Normal weight', 'Average');
  if (bmi < 30.0) return ('Overweight', 'Increased');
  if (bmi < 35.0) return ('Obese Class I', 'High');
  if (bmi < 40.0) return ('Obese Class II', 'Very High');
  return ('Obese Class III', 'Extremely High');
}

Classification classifyWhoAsian(double bmi) {
  if (bmi < 18.5) return ('Underweight', 'Low (but other risks)');
  if (bmi < 23.0) return ('Normal weight', 'Average');
  if (bmi < 27.5) return ('Overweight', 'Increased');
  return ('Obese', 'High');
}

Classification classifyPonderal(double pi) {
  if (pi < 11.0) return ('Underweight', 'Low');
  if (pi <= 14.0) return ('Normal', 'Average');
  return ('Overweight', 'Elevated');
}

Classification classifyNewBmi(double newBmi) => classifyWhoStandard(newBmi);

Classification classifyWhtr(double whtr) {
  if (whtr < 0.4) return ('Extremely Slim', 'Low (but other risks)');
  if (whtr < 0.5) return ('Healthy', 'Average');
  if (whtr < 0.6) return ('Increased Risk', 'Increased');
  return ('High Risk', 'High');
}

Classification classifyBodyFat(double bfPct, String sex) {
  if (sex == 'Male') {
    if (bfPct < 6)  return ('Essential Fat', 'Low (but other risks)');
    if (bfPct < 14) return ('Athletic', 'Average');
    if (bfPct < 18) return ('Fitness', 'Average');
    if (bfPct < 25) return ('Acceptable', 'Average');
    return ('Obese', 'High');
  } else {
    if (bfPct < 14) return ('Essential Fat', 'Low (but other risks)');
    if (bfPct < 21) return ('Athletic', 'Average');
    if (bfPct < 25) return ('Fitness', 'Average');
    if (bfPct < 32) return ('Acceptable', 'Average');
    return ('Obese', 'High');
  }
}
